mithril-common/src/entities/block_range.rs

```diff
@@ -14,23 +14,23 @@ use crate::{
 };
 
 /// BlockRangeLength is the length of a block range.
-pub type BlockRangeLength = u64;
+pub type BlockRangeLength = BlockNumber;
 
 /// BlockRange for the Cardano chain
 #[derive(Serialize, Deserialize, Clone, Eq, PartialEq, Debug, Hash)]
 pub struct BlockRange {
-    inner_range: Range<u64>,
+    inner_range: Range<BlockNumber>,
 }
 
 impl BlockRange {
     /// The length of the block range
     /// Important: this value should be updated with extreme care (probably with an era change) in order to avoid signing disruptions.
-    pub const LENGTH: BlockRangeLength = 15;
+    pub const LENGTH: BlockRangeLength = BlockNumber(15);
 
     /// BlockRange factory
-    pub fn new(start: BlockNumber, end: BlockNumber) -> Self {
+    pub fn new<T1: Into<BlockNumber>, T2: Into<BlockNumber>>(start: T1, end: T2) -> Self {
```

Comment:
```suggestion
    pub fn new<S: Into<BlockNumber>, E: Into<BlockNumber>>(start: S, end: E) -> Self {
```

Reply:
Generics are removed from `new` function.
`start`and `end` are u64 but this function is only available for test usage